### DISCUSSION TITLE
Add Shutdown, Reboot, Sleep commands as items to cinnamon app menu search.

### DISCUSSION THREAD

**Discussion by whoami-amrit** [linuxmint/discussions#55] [Ideas]
Hello! :wave:  I was wondering if `Shutdown`, `Reboot` & `Sleep` could be added as run commands to Cinnamon desktop app menu launcher.

Desired workflow -
- Press `Super` key followed by typing shutdown and pressing Enter. 

I would like to work on this as an MR. If possible please guide to a getting started with cinnamon development. 

This could be implemented as is or also be extended to allowing users to create custom functions/items for app launcher to run/execute.

**Comment by Secret-chest** [comment]
You can do it using the menu editor.

**Comment by rcalixte** [comment]
@Elessar1802 If you look into the "Run dialog" option under keyboard shortcuts (General -> Run dialog, the default is Alt + F2), there is an option to set a key binding for the run dialog. The run dialog supports terminal commands with autocomplete for the command names. While it would be a different key binding, does this work for you?

Another option like @Secret-chest said is that you can create new items using the Cinnamon Menu Editor (cinnamon-menu-editor) with the exact names and commands that you would like. I'm not sure if there would be an interest in a pull request for the .desktop files but at the very least, you could accomplish this for your setup(s).

**Reply by whoami-amrit** [reply]
Thanks! That works for me.
